lefthook.yml

```diff
@@ -0,0 +1,19 @@
+pre-commit:
+  parallel: true
+  commands:
+    spotless:
+      root: "backend/"
+      glob: "*.java"
+      run: docker run --rm -v ${PWD}:/app/backend prime-simplereport_backend sh -c "gradle spotlessApply" && git add {staged_files}
+    prettier:
+      root: "frontend/"
+      glob: "*.{js,ts,jsx,tsx}"
+      run: docker run --rm -v ${PWD}:/app/frontend prime-simplereport_frontend sh -c "npx prettier --config /app/frontend/package.json --write {staged_files}" && git add {staged_files}
```

Comment:
I am still not sure whats the advantage of running linting in docker. you still have to have `yarn`, `npm` and `lefthook` installed on local machine. might as well also install `prettier` `eslint` `stylelint` and run it directly on the machine vs having to rely on docker and installing it everytime via `npx`

Reply:
Yeah, you make a good point. I think it's a good idea to install the frontend linters locally since we already require node to be installed to run the pre-commit hooks. What do you think about the backend though (spotless)? I haven't required in my instructions to install java locally, but that would be needed to run a grade task locally. Would it make sense to still run that one in docker?

Reply:
I think that should also be kept outside docker and on the machine, i don't think its realistic to do app development on the backend without having a jdk installed locally since most IDE's will require it. Intellij requires jdk, it self installs it though. VS code will require it if you install the java extension, its a manual install though.

Reply:
I guess this sort of just calls the whole approach into question for me. Like, why bother running with docker if I _also_ have to install node and java on my local? At that point, it's just a few commands to have the app running on local. Maybe it would make more sense for the compose setup to just contain the database and nginx, and let the app itself run outside docker?

Reply:
I honestly think the docker approach to running it locally is valid, we need to have an easy official way to run the app locally.  An automated way to run both backend & frontend and supporting services. And the Docker setup you made is great for that! The linting is mostly automated and its not part of the `running` part of the development cycle.

Reply:
Can we roll all of these into a "test" Dockerfile? This file would have all of the dependencies needed for linting/testing, and it could use a local volume to store the dependencies, and then we would use that container for all of these steps. I personally would prefer to not maintain versions of multiple languages on my machine if I can help it.

Reply:
I think my concern is more surrounding dependency management. I think having a language runtime installed is required if you're developing in that language, so no problems there, but I'd like my dependency management of that language automated and abstracted away from me, if that makes sense.

Reply:
Awesome! lets do the test dockerfile approach